spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -883,6 +916,23 @@ private void checkIdle() {
 			}
 		}
 
+		private void idleBetweenPollIfNecessary() {
+			long idleBetweenPolls = this.containerProperties.getIdleBetweenPolls();
+			if (idleBetweenPolls > 0) {
+				idleBetweenPolls = Math.min(idleBetweenPolls,
+						this.maxPollInterval - (System.currentTimeMillis() - this.lastPoll));
+				if (idleBetweenPolls > 0) {
+					try {
+						TimeUnit.MILLISECONDS.sleep(idleBetweenPolls);
+					}
+					catch (InterruptedException ex) {
+						Thread.currentThread().interrupt();
+						throw new IllegalStateException("Consumer Thread [" + this + "] has been interrupted", ex);
```

Comment:
I don't think this is necessary and it will cause `wrapUp()` to be skipped because the call to this method is outside of the `handleConsumerException()` catch block.

Most likely an interrupt is because we're shutting down, but perhaps the safest thing to do is move the call to the try block so that `handleConsumerException()` will log this exception.